Objects/Objects/Geometry/Vector.cs

```diff
@@ -197,14 +196,28 @@ public static Vector CrossProduct(Vector u, Vector v)
     }
 
     /// <summary>
-    /// Divides this vector by it's euclidean length.
+    /// Compute and return a unit vector from this vector
+    /// </summary>
+    /// <returns>a normalized unit vector</returns>
+    public Vector Normalize()
+    {
+      var length = Length;
+      x /= length;
+      y /= length;
+      z /= length;
+      return this;
+    }
+
+    /// <summary>
+    /// Inverses the direction of the vector, equivalent to multiplying by -1
     /// </summary>
-    public void Unitize()
+    /// <returns>A pointing in the opposite direction</returns>
+    public Vector Negate()
     {
-      var length = this.Length;
-      this.x /= length;
-      this.y /= length;
-      this.z /= length;
+      x *= -1;
+      y *= -1;
+      z *= -1;
+      return this;
```

Comment:
Can we change this `Negate` method also to not `return this`?

Reply:
Hi @JR-Morgan, can help me explain why using this is not good for design API in this case? 
I've seen a lot of restrictions on not returning anything, which creates a lot of mess for developers using the API.
https://en.m.wikipedia.org/wiki/Fluent_interface
https://stackoverflow.com/questions/65173197/when-to-return-this-instead-of-void-in-a-method-and-why
https://stackoverflow.com/questions/781648/c-returning-this-for-method-nesting

This is a library I feel it good 
https://github.com/GSharker/G-Shark

Reply:
Great question,

There are a few reasons that come to mind.
1. We don't follow a fluent pattern for other Objects functions, so I don't see point in following it here.
2. imo, if I see a `public Vector Negate()` function, my first reaction is that this returns a copy of the object, that's negated.
    Which would not be the case here.
    
    In other contexts, like `GShark.Geometry.Vector3`, `System.Numerics.Vector`, `Rhino.Geometry.Vector3d`, `UnityEngine.Vector3`, an x,y,z vector is represented as a value type (struct), where returning a copy is preferred.
    
    But our `Vector` model is a reference type (class) because primarily it's a data transfer object.
So I'd like to avoid any sort of ambiguity about mutation vs copying.

Pinging @clairekuang in case they have any opinions.

Reply:
It mean will be improvement it in future or keep return void as a standard ? I readlly hard to use this API Objects Geometry at the moment. I need confirm it, thank you.